Packs/Base/Scripts/DBotFindSimilarIncidentsByIndicators/DBotFindSimilarIncidentsByIndicators.py

```diff
@@ -0,0 +1,507 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+from sklearn.base import BaseEstimator, TransformerMixin
+import pandas as pd
+import numpy as np
+from collections import Counter
+import re
+import math
+
+STATUS_DICT = {
+    0: "Pending",
+    1: "Active",
+    2: "Closed",
+    3: "Archive",
+}
+
+
+def normalize(x: List[str]) -> str:
+    """
+    Normalize function for indicators
+    :param x:  list of indicators
+    :return:
+    """
+    return ' '.join(x)
+
+
+def identity_score(x):
+    """
+    Identity function
+    :param x: object
+    :return:
+    """
+    return x
+
+
+def flatten(my_list: List[List]) -> List:
+    """
+    Flatten a list of list
+    :param l: list of list
+    :return: list
+    """
+    return [item for sublist in my_list for item in sublist]
+
+
+class Tfidf(BaseEstimator, TransformerMixin):
+    """
+    TFIDF class for indicator
+    """
+
+    def __init__(self, feature_names, tfidf_params, normalize_function, x):
+        self.feature_names = feature_names
+        self.params = tfidf_params
+        self.normalize_function = normalize_function
+        self.frequency = {}
+        if self.normalize_function:
+            x = x[self.feature_names].apply(self.normalize_function)
+        else:
+            x = x[self.feature_names]
+        self.vocabulary = x.iloc[0].split(' ')
+
+    def fit(self, x):
+        if self.normalize_function:
+            x = x[self.feature_names].apply(self.normalize_function)
+        else:
+            x = x[self.feature_names]
+        size = len(x) + 1
+        frequencies = Counter(flatten([t.split(' ') for t in x.values]))
+        frequencies.update(Counter(self.vocabulary))
+        self.frequency = {k: math.log(1 + size / v) for k, v in frequencies.items()}
+        return self
+
+    def transform(self, x):
+        if self.normalize_function:
+            x = x[self.feature_names].apply(self.normalize_function)
+        else:
+            x = x[self.feature_names]
+        return x.apply(self.compute_term_score)
+
+    def compute_term_score(self, x):
+        x = x.split(' ')
+        return sum([1 * self.frequency[word] for word in self.vocabulary if word in x]) / sum(
+            [self.frequency[word] for word in self.vocabulary])
+
+
+TRANSFORMATION = {
+    'indicators': {'transformer': Tfidf,
+                   'normalize': None,
+                   'params': {'analyzer': 'word', 'max_features': 200, 'token_pattern': '.'},  # [\d\D]*
+                   'scoring': {'scoring_function': identity_score, 'min': 0.5}
+                   }
+}
+
+
+class Transformer():
+    def __init__(self, p_type, field, p_incidents_df, p_incident_to_match, p_params):
```

Comment:
what is field? I think it's too generic name, I can't understand the meaning of this field

Reply:
you need to pass scoring function & min_score as class member, it's hard to what "p_params" contains